go/vt/topotools/routing_rules.go

```
@@ -158,4 +166,33 @@ func SaveKeyspaceRoutingRules(ctx context.Context, ts *topo.Server, rules map[st
 	return ts.SaveKeyspaceRoutingRules(ctx, keyspaceRoutingRules)
 }
 
+// UpdateKeyspaceRoutingRules updates the keyspace routing rules in the topo server. It initially acquires a
+// RoutingRulesLock and then calls the update function to modify the rules in-place.
+// If the update function returns an error, the rules are not saved and the lock is released.
+// If the update function is successful, the rules are saved to the topo and the lock is released.
+func UpdateKeyspaceRoutingRules(ctx context.Context, ts *topo.Server, reason string,
+	update func(ctx context.Context, rules *map[string]string) error) (err error) {
+	var lock *topo.RoutingRulesLock
+	lock, err = topo.NewRoutingRulesLock(ctx, ts, reason)
+	if err != nil {
+		return err
+	}
+	lockCtx, unlock, lockErr := lock.Lock(ctx)
+	if lockErr != nil {
+		return lockErr
+	}
+	defer unlock(&err)
+	rules, _ := GetKeyspaceRoutingRules(lockCtx, ts)
```

If we do keep this code, we don't want to ignore the error here do we?